### ISSUE TITLE
Option to hold L/R swap button instead of tapping it

### ISSUE THREAD

**Issue by omgitsraven** [chrisboyle/sgtpuzzles#493]
For games where I'm switching frequently (i.e. Slant), I would like if there were an option to use the L/R swap button as follows:

1. Press my finger down on swap button (and leave it there)
2. Press my finger down on a cell, then lift it up again: R touch is performed
3. Lift my finger up from swap button
4. Press my finger down on another cell, then lift it up again: L touch is performed.

i.e. an option for the swap button to only be active while my finger is STILL ON it, rather than for it to be a permanent swap until the next time I touch it.

Maybe if a touch/release happens on the swap WITHOUT touching the board, it could continue to behave as before, so the 'default' state could still be easily changed, for games with a sort of 'second phase' where one state becomes much more likely than the other but frequent swapping is still somewhat required?

To summarize, I would like the swap button to behave as follows:
* It changes state immediately upon touch START, rather than touch end
* Upon touch end:
  * if the player has changed the puzzle state with another finger during this touch, the swap setting is now changed BACK
  * if the player has NOT changed the puzzle state during this touch, the new swap setting acquired during the touch start is persisted

i.e. the button's new functionality will only make a difference if people are performing multi-touches, so existing players won't be inconvenienced.
